helix-core/src/main/java/org/apache/helix/controller/rebalancer/waged/AssignmentMetadataStore.java

```diff
@@ -92,11 +92,14 @@ protected AssignmentMetadataStore(BucketDataAccessor bucketDataAccessor, String
     return _bestPossibleAssignment;
   }
 
-  public void persistBaseline(Map<String, ResourceAssignment> globalBaseline) {
+  /**
+   * @return true if a new baseline was persisted.
+   */
+  public boolean persistBaseline(Map<String, ResourceAssignment> globalBaseline) {
     // TODO: Make the write async?
     // If baseline hasn't changed, skip writing to metadata store
     if (compareAssignments(_globalBaseline, globalBaseline)) {
```

Comment:
Can we move this check outside of this method? My 2 cents, this persisting method only does persisting. True if successful, false if fails. If baseline is not changed, it should be decided outside of this method. If you call this method to persist, I will Do my job to persist it no matter it is not changed or not.

Reply:
Please refer to Hunter's comments. Let's discuss the same problem in the same thread.